src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/QuestionManager.java

```diff
@@ -619,4 +629,53 @@ public ChoiceDTO convertJsonAnswerToChoice(final String jsonAnswer) throws Error
     }
     return answerFromClientDTO;
   }
+
+  public GameFilter createGameFilterForRandomQuestions(final RegisteredUserDTO currentUser, final String subjects) {
+    GameFilter gameFilter = new GameFilter();
+
+    UserPreference filterQuestionsPreference = null;
+    try {
+      filterQuestionsPreference = this.userPreferenceManager.getUserPreference(
+          "DISPLAY_SETTING",
+          "HIDE_NON_AUDIENCE_CONTENT",
+          currentUser.getId());
+    } catch (SegueDatabaseException e) {
+      SegueErrorResponse error = new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR,
+          "Error while getting user preferences", e);
+      log.error(error.getErrorMessage(), e);
+    }
+
+    if (filterQuestionsPreference != null && filterQuestionsPreference.getPreferenceValue()) {
+      var userContexts = currentUser.getRegisteredContexts();
+
+      List<String> subjectsList = splitCsvStringQueryParam(subjects);
+      List<String> stagesList = new ArrayList<>();
+      List<String> examBoardsList = new ArrayList<>();
+
+      for (UserContext uc : userContexts) {
+        stagesList.add(uc.getStage().name());
+        examBoardsList.add(uc.getExamBoard().name());
+      }
+
+      gameFilter = new GameFilter(
+          subjectsList,
+          null,
+          null,
+          null,
+          null,
+          null,
+          stagesList,
+          null,
+          examBoardsList);
+    }
+    return gameFilter;
+  }
+
+  private static List<String> splitCsvStringQueryParam(final String queryParamCsv) {
```

Comment:
This is almost a carbon copy of another existing function, so we shouldn't be duplicating code like this.

Reply:
Where would you like this to be added, or do you want me to add another dependency, this time to "GameboardsFacade.java", or do you want me to move one of the implementations somewhere else, add the dependency to both files and remove the unused one?

Reply:
What I would do in a case like this is to extract the original function out to some other file, perhaps there is an existing "utils" class already, or a "CSV utils" or "query utils" class.
The only problem now is that both functions are slightly different (because you implemented my recommendation, before I realised it was a copy/pasted function). So I would simply ignore my recommendation and use the implementation of the original function, just to make sure that we don't introduce any breaking changes.